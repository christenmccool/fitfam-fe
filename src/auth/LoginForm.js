import React, {useState} from 'react';
import { useNavigate } from 'react-router-dom';

import Box from '@mui/material/Box';
import Container from '@mui/material/Container';
import Typography from '@mui/material/Typography';
import TextField from '@mui/material/TextField';
import Button from '@mui/material/Button';

import Alert from '../common/Alert';

/** LoginForm
 *
 * Form for user login given email and password 
 * 
 * Routed at /login
 */
const LoginForm = ({ login }) => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [errors, setErrors] = useState([]);
  const navigate = useNavigate();

  const handleEmailChange = (event) => {
    const {value} = event.target;
    setEmail(value);
  };

  const handlePasswordChange = (event) => {
    const {value} = event.target;
    setPassword(value);
  };

  const handleSubmit = async (event) => {
    event.preventDefault();

    const result = await login(email, password);
    if (result.success) {
      navigate('/');
    } else {
      setErrors(result.err);
    } 
  };


  return (
    <Container align="center" maxWidth="sm" sx={{backgroundColor: "#FFF", borderRadius: '10px'}}>
      <Box m={5} py={2}>
        <Typography component="h1" variant="h4" textAlign="center" color="primary">
          Login to FitFam
        </Typography>

        <Box component="form" onSubmit={handleSubmit} mt={2}>
          <TextField
            margin="normal"
            fullWidth
            required
            id="email"
            name="email"
            label="Email Address"
            autoFocus
            onChange={handleEmailChange}
            value={email}
          />
          <TextField
            margin="normal"
            fullWidth
            required
            id="password"
            name="password"
            label="Password"
            type="password"
            onChange={handlePasswordChange}
            value={password}
          />
          <Button
            type="submit"
            variant="contained"
            size="large"
            sx={{ mt: 3, mb: 2 }}
          >
            Login
          </Button>
        </Box>

        {errors.length ?
          <Alert messages={errors} />
          : null
        }
      </Box>
    </Container>
  )
}

export default LoginForm;