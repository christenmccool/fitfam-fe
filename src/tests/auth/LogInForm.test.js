import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import UserEvent from "@testing-library/user-event";

import LoginForm from "../../auth/LoginForm";


it('renders without crashing', () => {
  render(
    <MemoryRouter>
      <LoginForm />
    </MemoryRouter>
  );
});


it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <LoginForm />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});


it("renders form", function () {
  render(
    <MemoryRouter>
      <LoginForm />
    </MemoryRouter>
  );
  expect(screen.getByRole('textbox', {name: /email/i})).toBeInTheDocument();
  expect(screen.getByLabelText(/password/i)).toBeInTheDocument();
});


it("responds to user input", function () {
  render(
    <MemoryRouter>
      <LoginForm />
    </MemoryRouter>
  );

  const emailField = screen.getByRole('textbox', {name: /email/i});
  UserEvent.type(emailField, "test@mail.com");
  expect(emailField.value).toEqual("test@mail.com");  

  const passwordField = screen.getByLabelText(/password/i);
  UserEvent.type(passwordField, "testpassword");
  expect(passwordField.value).toEqual("testpassword");  
});

