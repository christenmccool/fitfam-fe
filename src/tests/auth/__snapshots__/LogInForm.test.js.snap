// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    align="center"
    class="MuiContainer-root MuiContainer-maxWidthSm css-ead4d9-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-1yciqgl"
    >
      <h1
        class="MuiTypography-root MuiTypography-h4 css-1oih086-MuiTypography-root"
      >
        Login to FitFam
      </h1>
      <form
        class="MuiBox-root css-1yuhvjn"
      >
        <div
          class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused Mui-required css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="email"
            id="email-label"
          >
            Email Address
            <span
              aria-hidden="true"
              class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               *
            </span>
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="email"
              name="email"
              required=""
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-7nxzgd"
              >
                <span>
                  Email Address *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="password"
            id="password-label"
          >
            Password
            <span
              aria-hidden="true"
              class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               *
            </span>
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="password"
              name="password"
              required=""
              type="password"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1g8cwq9"
              >
                <span>
                  Password *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-isfcrp-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Login
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
